app/Console/Command/AddLabelToItems.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Console\Command;
+
+use Broadway\CommandHandling\CommandBus;
+use CultuurNet\UDB3\Model\ValueObject\Identity\ItemType;
+use CultuurNet\UDB3\Model\ValueObject\Taxonomy\Label\Label;
+use CultuurNet\UDB3\Model\ValueObject\Taxonomy\Label\LabelName;
+use CultuurNet\UDB3\Offer\Commands\AddLabel as OfferAddLabel;
+use CultuurNet\UDB3\Organizer\Commands\AddLabel as OrganizerAddLabel;
+use CultuurNet\UDB3\Search\ResultsGenerator;
+use CultuurNet\UDB3\Search\SearchServiceInterface;
+use CultuurNet\UDB3\Search\Sorting;
+use Exception;
+use Generator;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+final class AddLabelToItems extends AbstractCommand
+{
+    private ResultsGenerator $eventsSearchResultsGenerator;
+
+    private ResultsGenerator $placesSearchResultsGenerator;
+
+    private ResultsGenerator $organizersSearchResultsGenerator;
+
+    public function __construct(
+        CommandBus $commandBus,
+        SearchServiceInterface $eventsSearchService,
+        SearchServiceInterface $placesSearchService,
+        SearchServiceInterface $organizersSearchService
+    ) {
+        $this->eventsSearchResultsGenerator = new ResultsGenerator(
```

Comment:
I think you can make an internal method that creates the generator based on the item type. Once you have this method you can use it inside the execute and avoid a lot of duplicated code.

Reply:
The only thing that is different seems to be the injected search service. So a lot of duplication can be avoided, even the sorting and page size is the same.